include/oneapi/dpl/internal/dynamic_selection_impl/scoring_policy_defs.h

```diff
@@ -0,0 +1,56 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+
+#ifndef _SCORING_POLICY_DEFS_H
+#define _SCORING_POLICY_DEFS_H
+
+#pragma once
+
+namespace oneapi {
+namespace dpl {
+namespace experimental {
+  struct nop_property_handle_t {
+    static constexpr bool should_report_task_submission = false;
+    static constexpr bool should_report_task_completion = false;
+    static constexpr bool should_report_task_execution_time = false;
+    template<typename Property>
+    static constexpr void report(const Property&) noexcept { return; }
+    template<typename Property, typename ValueType>
+    static constexpr void report(const Property&, ValueType) noexcept { return; }
+  };
+  inline constexpr nop_property_handle_t nop_property_handle;
```

Comment:
Same for the no-op property handle and selection handle classes - what are these for?

Reply:
Replaced all nop_* classes with basic